rules.bolt

```diff
@@ -0,0 +1,124 @@
+hasRole(role) { auth.token[role] === true }
+
+type CurrentTimestamp extends Number {
+  validate() { this == now }
+}
+
+type InitialTimestamp extends CurrentTimestamp {
+  validate() { prior(this) == null }
+}
+
+type ReadOnlyCoordinator {
+  read() { hasRole("coordinator") }
+}
+
+type WriteOnlyCoordinator {
+  write() { hasRole("coordinator") }
+}
+
+type OnlyNew {
+  validate() { prior(this) == null }
+}
+
+type EditStatus extends WriteOnlyCoordinator {
+  validate() {
+    this.test(
+      /"Spare" | "Given" | "Submitted" | "Revise" | "Done"/
+    )
+  }
+}
+
+/donations {
+  /cash is ReadOnlyCoordinator {
+    path /{push_id} {
+      create() {
+        return root.donations.tokens[this.token].valid == true
+      }
+    }
+  }
+}
+
+/original is ReadOnlyCoordinator {
+  /{list} {
+    index() { ["soundQualityReporting/status"] }
+    /{fileName} {
+      /contentReporting {
```

Comment:
This whole block repeats below as well. And it is going to be the same for track editing also. Can we extract some common block?